invokeai/backend/quantization/gguf/torch_patcher.py

```diff
@@ -0,0 +1,90 @@
+# Largely based on https://github.com/city96/ComfyUI-GGUF
+
+from contextlib import contextmanager
+from typing import Generator, Optional
+
+import wrapt
+from torch import Tensor, bfloat16, dtype, float16, nn
+
+from invokeai.backend.quantization.gguf.layers import GGUFLayer
+
+
+class TorchPatcher:
+    @classmethod
+    @contextmanager
+    def wrap(cls) -> Generator[None, None, None]:
+        # Dictionary to store original torch.nn classes for later restoration
+        original_classes = {}
+        try:
+            # Iterate over _patcher's attributes and replace matching torch.nn classes
+            for attr_name in dir(cls):
+                if attr_name.startswith("__"):
+                    continue
+                # Get the class from _patcher
+                patcher_class = getattr(cls, attr_name)
+
+                # Check if torch.nn has a class with the same name
+                if hasattr(nn, attr_name):
+                    # Get the original torch.nn class
+                    original_class = getattr(nn, attr_name)
+
+                    # Define a helper function to bind the current patcher_attr for each iteration
+                    def create_patch_function(patcher_attr):
+                        # Return a new patch_class function specific to this patcher_attr
+                        @wrapt.decorator
+                        def patch_class(wrapped, instance, args, kwargs):
+                            # Call the _patcher version of the class
+                            return patcher_attr(*args, **kwargs)
+
+                        return patch_class
+
+                    # Save the original class for restoration later
+                    original_classes[attr_name] = original_class
+
+                    # Apply the patch
+                    setattr(nn, attr_name, create_patch_function(patcher_class)(original_class))
```

Comment:
And here pyright is complaining about calling the untyped `create_patch_function` in a typed context. `create_patch_function()` should be returning a Callable, but I wasn't sure of the exact signature needed.